client/client.go

```diff
@@ -493,9 +497,22 @@ func (c *Client) ListImages(ctx context.Context, filters ...string) ([]Image, er
 	if err != nil {
 		return nil, err
 	}
-	images := make([]Image, len(imgs))
-	for i, img := range imgs {
-		images[i] = NewImage(c, img)
+
+	// With image pull per runtime class feature, an image can now exist
+	// for different platforms. Therefore, look at the image platform label
+	// and set the platform matcher for the image accordingly.
+	var listOfPlatformsForImage []string
+	var images []Image
+	for _, img := range imgs {
+		for key, value := range img.Labels {
+			if strings.HasPrefix(key, ctrdlabels.PlatformLabelPrefix) {
+				listOfPlatformsForImage = append(listOfPlatformsForImage, value)
+			}
+		}
+
+		for _, platform := range listOfPlatformsForImage {
+			images = append(images, NewImageWithPlatform(c, img, platforms.Only(platforms.MustParse(platform))))
```

Comment:
I think it's not a good to change basic API logic here. For containerd API, the image service already supports multiple platforms. It doesn't make sense to change here. We should add a wrapper function in CRI plugin side.